src/Microsoft.Performance.SDK.Runtime/AssemblyTypeReferenceWithInstance.cs

```diff
@@ -42,13 +62,62 @@ protected AssemblyTypeReferenceWithInstance(Type type)
             AssemblyTypeReferenceWithInstance<T, Derived> other)
             : base(other)
         {
+            this.instanceFactory = other.instanceFactory;
             this.Instance = other.Instance;
         }
 
+        /// <summary>
+        ///     Finalizes an instance of the <see cref="AssemblyTypeReferenceWithInstance{T, Derived}"/>
+        ///     class.
+        /// </summary>
+        ~AssemblyTypeReferenceWithInstance()
+        {
+            this.Dispose(false);
+        }
+
         /// <summary>
         ///     Gets an instance of <see cref="T"/>
         /// </summary>
-        public T Instance { get; }
+        /// <exception cref="ObjectDisposedException">
+        ///     This instance is disposed.
+        /// </exception>
+        public T Instance
+        {
+            get
+            {
+                this.ThrowIfDisposed();
+                return this.instance;
+            }
+            private set
+            {
+                this.ThrowIfDisposed();
+                this.instance = value;
+            }
+        }
+
+        /// <summary>
+        ///     Disposes the <see cref="Instance"/> referenced by this instance 
+        ///     without disposing this instance. The <see cref="Instance"/> is
```

Comment:
wording here feels a bit ambiguous. perhaps replace "instance" (not the property) with "object"?

Reply:
After speaking more offline, we realized this code isn't needed right now, so I am just removing it.